internal/server/health.go

```diff
@@ -0,0 +1,27 @@
+package server
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+type healthResponse struct {
+	CommitHash   string `json:"commit_hash"`
+	BuildVersion string `json:"build_version"`
+	BuildDate    string `json:"build_date"`
+	GoVersion    string `json:"go_version"`
+}
+
+func (s *Server) healthHandler(w http.ResponseWriter, r *http.Request) {
+	err := json.NewEncoder(w).Encode(healthResponse{
+		CommitHash:   CommitHash,
+		BuildVersion: BuildVersion,
+		BuildDate:    BuildDate,
+		GoVersion:    GoVersion,
+	})
+	if err != nil {
+		s.logger.Printf("failed to write health response: %v\n", err)
```

Comment:
Nit:
```suggestion
		s.logger.Printf("failed to write health response: %w\n", err)
```

Reply:
This is fine. We are logging it. Not sending it upwards.